File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/fast/EcoFastItemStack.java
package com.willfp.eco.internal.fast;

import com.willfp.eco.core.fast.FastItemStack;
import lombok.Getter;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public abstract class EcoFastItemStack<T> implements FastItemStack {
    @Getter
    private final T handle;

    @Getter
    private final ItemStack bukkit;

    protected EcoFastItemStack(@NotNull final T handle,
                               @NotNull final ItemStack bukkit) {
        this.handle = handle;
        this.bukkit = bukkit;
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/fast/EcoFastItemStack.kt
package com.willfp.eco.internal.fast

import com.willfp.eco.core.fast.FastItemStack
import org.bukkit.inventory.ItemFlag
import org.bukkit.inventory.ItemStack

abstract class EcoFastItemStack<T: Any>(
    val handle: T,
    val bukkit: ItemStack
) : FastItemStack {
    override fun unwrap(): ItemStack {
        return bukkit
    }

    fun getBitModifier(hideFlag: ItemFlag): Int {
        return 1 shl hideFlag.ordinal
    }
}